frontend/kubecloud/public/docs/tutorial.md

```diff
@@ -912,241 +363,171 @@ spec:
+                  self.wfile.write(json.dumps(response).encode())
+          with socketserver.TCPServer(("", 8002), Handler) as httpd:
+              httpd.serve_forever()
+        ports:
+        - containerPort: 8002
 ```
 
-**Kubeconfig / Access Issues:**
-
-```bash
-# Show current context and clusters
-kubectl config view --minify
-kubectl config get-contexts
-kubectl config use-context <context>
+### Step 4 — Services (save as `microservices-svc.yaml`)
 
-# Verify KUBECONFIG path (if using a custom location)
-echo $KUBECONFIG   # (Linux/macOS)
-$Env:KUBECONFIG    # (Windows PowerShell)
-
-# Inspect certificate expiration (if applicable)
-kubectl -n kube-system get secrets | grep kube
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: frontend-service
+spec:
+  selector:
+    app: microservice
+    service: frontend
+  ports:
+  - port: 8000
+    targetPort: 8000
+  type: LoadBalancer
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: backend-service
+spec:
+  selector:
+    app: microservice
+    service: backend
+  ports:
+  - port: 8001
+    targetPort: 8001
+  type: ClusterIP
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: database-service
+spec:
+  selector:
+    app: microservice
+    service: database
+  ports:
+  - port: 8002
+    targetPort: 8002
+  type: ClusterIP
 ```
 
-**Storage (PV/PVC) Issues:**
+### Step 5 — Apply and test
 
 ```bash
-# Check PVC and PV status
-kubectl get pvc -A
-kubectl get pv
-
-# Describe problematic claims
-kubectl describe pvc -n <namespace> <pvc-name>
-
-# Check StorageClass and provisioner
-kubectl get storageclass
-kubectl describe storageclass <name>
+kubectl apply -f frontend.yaml
+kubectl apply -f backend.yaml
+kubectl apply -f database.yaml
+kubectl apply -f microservices-svc.yaml
+kubectl port-forward service/frontend-service 8080:8000
 ```
 
-**Ingress / DNS Issues:**
-
-```bash
-# Verify ingress resources
-kubectl get ingress -A
-kubectl describe ingress -n <namespace> <name>
-
-# Test service reachability inside the cluster
-kubectl run tmp --image=busybox:1.36 --restart=Never -it --rm -- sh -c "wget -qO- http://<service>.<namespace>.svc.cluster.local:<port>"
-
-# Validate DNS resolution from your workstation
-nslookup <your-domain>
-curl -v https://<your-domain>
-```
+Open `http://localhost:8080` — the frontend should call the backend and database over Mycelium networking.
 
-**Collect Diagnostics Quickly:**
+### Step 6 — Verify Communication
 
 ```bash
-# Save a snapshot of cluster state (non-sensitive)
-kubectl cluster-info dump --all-namespaces --output-directory=./cluster-dump
+# Check which nodes pods are running on
+kubectl get pods -o wide
 
-# Events sorted by time
-kubectl get events -A --sort-by=.metadata.creationTimestamp | tail -n 200
+# Test service communication
+kubectl exec -it deployment/frontend-service -- curl http://backend-service:8001/api/data
+kubectl exec -it deployment/backend-service -- curl http://database-service:8002/api/stats
```

Comment:
i can't run this command, looks like the container do not have curl on it 

kubectl exec -it deployment/frontend-service -- curl http://backend-service:8001/api/data 
error: Internal error occurred: Internal error occurred: error executing command in container: failed to exec in container: failed to start exec "cb99fdd38d6daec19dcde96f44258e7936f66e31a5ef85f2cc37d8c24f887a04": OCI runtime exec failed: exec failed: unable to start container process: exec: "curl": executable file not found in $PATH


however after installing the curl on the container it works fine